File: debug_mobile_scroll.mjs
import { chromium, devices } from 'playwright';

async function run() {
    const browser = await chromium.launch({ headless: true });
    const context = await browser.newContext(devices['iPhone 12']);
    const page = await context.newPage();

    page.on('console', msg => console.log(`BROWSER LOG: ${msg.text()}`));

    console.log("Navigating to http://localhost:3000...");
    await page.goto("http://localhost:3000", { waitUntil: 'networkidle' });
    
    await new Promise(resolve => setTimeout(resolve, 2000));

    console.log("Looking for Contact tile...");
    const contactTile = page.getByText("Contact", { exact: true });
    
    if (await contactTile.isVisible()) {
        console.log("Clicking Contact tile...");
        await contactTile.click();
        
        console.log("Waiting for scroll sequence to complete (6 seconds)...");
        await new Promise(resolve => setTimeout(resolve, 6000));
        
        const scrollY = await page.evaluate(() => {
            const container = document.querySelector('div.overflow-y-auto');
            return container ? container.scrollTop : -1;
        });
        console.log(`Final scroll position: ${scrollY}`);
        
        await page.screenshot({ path: "mobile_contact_scroll.png" });
        console.log("Screenshot saved to mobile_contact_scroll.png");
    } else {
        console.log("Contact tile not found!");
        await page.screenshot({ path: "debug_not_found.png" });
    }

    await browser.close();
}

run().catch(err => {
    console.error(err);
    process.exit(1);
});

File: debug_mobile_scroll.py
from playwright.sync_api import sync_playwright
import time

def run():
    with sync_playwright() as p:
        # iPhone 12 as a representative mobile device
        iphone_12 = p.devices['iPhone 12']
        browser = p.chromium.launch(headless=True)
        context = browser.new_context(**iphone_12)
        page = context.new_page()

        # Listen for console logs
        page.on("console", lambda msg: print(f"BROWSER LOG: {msg.text}"))

        print("Navigating to http://localhost:3000...")
        page.goto("http://localhost:3000", wait_until="networkidle")
        
        # Wait a bit for any hydration
        time.sleep(2)

        print("Looking for Contact tile...")
        # Search for the text "Contact" in the specific div structure of mobile tiles
        contact_tile = page.get_by_text("Contact", exact=True)
        
        if contact_tile.is_visible():
            print("Clicking Contact tile...")
            contact_tile.click()
            
            # Wait for the scroll animation to complete (we have 800ms delay + 2500ms duration)
            print("Waiting for scroll sequence to complete (4 seconds)...")
            time.sleep(4)
            
            # Check final scroll position
            scroll_y = page.evaluate("document.querySelector('div.overflow-y-auto').scrollTop")
            print(f"Final scroll position: {scroll_y}")
            
            # Take a screenshot to verify
            page.screenshot(path="mobile_contact_scroll.png")
            print("Screenshot saved to mobile_contact_scroll.png")
        else:
            print("Contact tile not found!")
            # Print page content or take screenshot for debugging
            page.screenshot(path="debug_not_found.png")

        browser.close()

if __name__ == "__main__":
    run()
